### DISCUSSION TITLE
Schema for catching RegExp

### DISCUSSION THREAD

**Discussion by Mikiri1337** [colinhacks/zod#3167] [Q&A] [answered]
Version: 3.22.4

Hello, everyone! I need help with creating schema for catching RegExp in any place of object (object can be infinite nested and have any keys as object/array/primitive).

```typescript
const obj = {
    arr: [{ prop: /\d/ }]
}
// How to catch RegExp at arr[0].prop?
```

**Comment by JacobWeisenburger** [comment, marked as answer]
It seems like you are trying to do something that I have never needed to do. Perhaps there is a way to simplify the problem. If you would like me to take a deep dive into your issue and try to come up with a solution, it will require a [Sponsored Consulting Session](https://github.com/sponsors/JacobWeisenburger?frequency=one-time). Let me know if that's something you would be interested in.
